src/bundle.ts

```diff
@@ -14,45 +14,75 @@
 
 import File = require('vinyl');
 import * as parse5 from 'parse5';
-import {Analyzer} from 'polymer-analyzer';
-import {Bundler} from 'polymer-bundler';
-import {BundleManifest, BundleStrategy, generateShellMergeStrategy} from 'polymer-bundler/lib/bundle-manifest';
+import {Bundler, Options as BundlerOptions} from 'polymer-bundler';
+import {BundleManifest, BundleStrategy, generateShellMergeStrategy, BundleUrlMapper} from 'polymer-bundler/lib/bundle-manifest';
 import {ProjectConfig} from 'polymer-project-config';
 import {Transform} from 'stream';
 
 import {BuildAnalyzer} from './analyzer';
 import {FileMapUrlLoader} from './file-map-url-loader';
 import {pathFromUrl, urlFromPath} from './path-transformers';
 
+export type Options =
+    BundlerOptions & {strategy?: BundleStrategy, urlMapper?: BundleUrlMapper};
+
 export class BuildBundler extends Transform {
   config: ProjectConfig;
 
   private _buildAnalyzer: BuildAnalyzer;
   private _bundler: Bundler;
 
+  private _strategy?: BundleStrategy;
+  private _urlMapper?: BundleUrlMapper;
+
   // A map of urls to file objects.  As the transform stream handleds files
   // coming into the stream, it collects all files here.  After bundlling,
   // we remove files from this set that have been inlined and replace
   // entrypoint/fragment files with bundled versions.
   files = new Map<string, File>();
 
-  constructor(config: ProjectConfig, buildAnalyzer: BuildAnalyzer) {
+  constructor(
+      config: ProjectConfig,
+      buildAnalyzer: BuildAnalyzer,
+      options?: Options) {
     super({objectMode: true});
 
+    if (!options) {
+      options = {};
+    }
+
     this.config = config;
 
     this._buildAnalyzer = buildAnalyzer;
-    this._bundler = new Bundler({
-
-      // TODO(usergenic): Creating a new Analyzer with a blank cache is going
-      // to mean, at least, a doubling of analysis efforts for bundling phase.
-      // Ideally we would fork existing analyzer and replace its urlLoader if
-      // there were an affordance to do so.
-      analyzer: new Analyzer(
-          {urlLoader: new FileMapUrlLoader(this.config.root, this.files)}),
-      inlineCss: true,
-      inlineScripts: true,
-    });
+
+    let {analyzer,
+         excludes,
+         inlineCss,
+         inlineScripts,
+         rewriteUrlsInTemplates,
+         sourcemaps,
+         stripComments,
+         strategy,
+         urlMapper} = options;
+
+    const urlLoader = new FileMapUrlLoader(
+        this.config.root, this.files, analyzer || buildAnalyzer.analyzer);
```

Comment:
I'm doing essentially the identical thing that the `InMemoryOverlayUrlLoader` does in Analyzer; here I just have need to use `File` objects as opposed to content strings which is why I'm not using the former.  Both loaders require a fallback, and since I can't fallback to the previous loader, because it is not exposed, I have to fallback to the previous Analyzer, which implements it.

However, you raise a fair point about the variable naming in that reassigning analyzer may be obfuscating what's happening.  I've adjusted that.